ClosedXML/Excel/PivotTables/XLPivotTable.cs

```diff
@@ -42,7 +44,65 @@ public IEnumerable<string> SourceRangeFieldsAvailable
 
         public IXLPivotTable SetTheme(XLPivotTableTheme value) { Theme = value; return this; }
 
-        public String Name { get; set; }
+        private Boolean ValidateName(String newName, String oldName, out String message)
+        {
+            message = "";
+            if (String.IsNullOrWhiteSpace(newName))
+            {
+                message = $"The table name '{newName}' is invalid";
+                return false;
+            }
+
+            // Table names are case insensitive
+            if (!oldName.Equals(newName, StringComparison.OrdinalIgnoreCase)
+                && Worksheet.Tables.Any(t => t.Name.Equals(newName, StringComparison.OrdinalIgnoreCase)))
```

Comment:
`Worksheet.Tables.Contains(newName)` would be better as it uses Dictionary internally.

Reply:
I moved the method to `XLHelper`. Tables, pivot tables and named ranges now all refer to one single method.